geode-core/src/integrationTest/java/org/apache/geode/pdx/JSONFormatterBasicJUnitTest.java

```diff
@@ -99,7 +105,62 @@ public void emptyObjectParses() {
   }
 
   @Test
-  public void simpleObjectParses() {
-    JSONFormatter.fromJSON("{\"a\": 2}");
+  @Parameters({"true", "false"})
+  public void simpleObjectAsStringParsesWithIdentityField(String usePdxInstanceSortedHelper) {
+    String originalSortJsonFieldNamesProperty =
+        System.getProperty(JSONFormatter.SORT_JSON_FIELD_NAMES_PROPERTY);
+    try {
+      System.setProperty(JSONFormatter.SORT_JSON_FIELD_NAMES_PROPERTY, usePdxInstanceSortedHelper);
+      String identityField = "a";
+      String nonIdentityField = "b";
+      String nonExistentField = "c";
+      String jsonObjectString = "{\"a\":2,\"b\":3}";
+
+      PdxInstance pdxInstance = JSONFormatter.fromJSON(jsonObjectString, identityField);
+      String deserializedJsonObjectString = JSONFormatter.toJSON(pdxInstance);
+
+      assertThat(pdxInstance.isIdentityField(identityField)).isTrue();
+      assertThat(pdxInstance.isIdentityField(nonIdentityField)).isFalse();
+      assertThat(pdxInstance.isIdentityField(nonExistentField)).isFalse();
+      assertThat(pdxInstance.hasField(identityField)).isTrue();
+      assertThat(pdxInstance.hasField(nonIdentityField)).isTrue();
+      assertThat(pdxInstance.hasField(nonExistentField)).isFalse();
+      assertThat(deserializedJsonObjectString).isEqualTo(jsonObjectString);
+    } finally {
+      if (originalSortJsonFieldNamesProperty != null) {
+        System.setProperty(JSONFormatter.SORT_JSON_FIELD_NAMES_PROPERTY,
```

Comment:
You should use the RestoreSystemProperties rule.  It'll make sure things are put back the way they should be.  As it stands it looks like you'll leave this property set if it wasn't set to something else before this test is run.

Reply:
Thanks Bruce - will do.